File: sort/insertion_sort/InsertionSort.java
public class InsertionSort {
    public static void main(String[] args) {
        // run with -ea (enable assertions) for jvm
        int[] array = new int[] {9, 6, 10, 8, 5, 3, 23, 99, 5, 62};
        int[] expectedArray = new int[] {3, 5, 5, 6, 8, 9, 10, 23, 62, 99};

        insertionSort(array);

        for (int i = 0; i < array.length; i++) {
            assert(array[i] == expectedArray[i]);
        }
    }

    private static void insertionSort(int[] array) {
        for (int i = 1; i < array.length; i++) {
            int key = array[i];
            int j = i - 1;

            while (j >= 0 && array[j] > key) {
                array[j + 1] = array[j--];
            }
            array[j + 1] = key;
        }
    }
}

File: sort/insertion_sort/insertion_sort.py
from typing import List


def insertionSort(A: List[int]) -> None:
    for i in range(1, len(A)):
        key = A[i]
        j = i - 1
        while j >= 0 and A[j] > key:
            A[j + 1] = A[j]
            j -= 1

        A[j + 1] = key


if __name__ == "__main__":
    array = [9, 6, 10, 8, 5, 3, 23, 99, 5, 62]
    expectedArray = [3, 5, 5, 6, 8, 9, 10, 23, 62, 99]

    insertionSort(array)

    assert array == expectedArray